include/sys/dmu_zfetch.h

```diff
@@ -49,19 +49,22 @@ typedef struct zfetch {
 
 typedef struct zstream {
 	uint64_t	zs_blkid;	/* expect next access at this blkid */
-	uint64_t	zs_pf_blkid;	/* next block to prefetch */
+	uint64_t	zs_pf_blkid1;	/* first block to prefetch */
+	uint64_t	zs_pf_blkid;	/* block to prefetch up to */
 
 	/*
 	 * We will next prefetch the L1 indirect block of this level-0
 	 * block id.
 	 */
-	uint64_t	zs_ipf_blkid;
+	uint64_t	zs_ipf_blkid1;	/* first block to prefetch */
+	uint64_t	zs_ipf_blkid;	/* block to prefetch up to */
 
-	kmutex_t	zs_lock;	/* protects stream */
+	list_node_t	zs_node;	/* link for zf_stream */
 	hrtime_t	zs_atime;	/* time last prefetch issued */
 	hrtime_t	zs_start_time;	/* start of last prefetch */
-	list_node_t	zs_node;	/* link for zf_stream */
 	zfetch_t	*zs_fetch;	/* parent fetch */
+	boolean_t	zs_missed;	/* stream saw cache misses */
+	zfs_refcount_t	zs_callers;	/* number of pending callers */
 	zfs_refcount_t	zs_blocks; /* number of pending blocks in the stream */
```

Comment:
I think `zs_blocks` now includes additional counts besides the pending blocks.  Is it similar to `number of pending blocks plus number of pending callers`?

Reply:
I've renamed zs_blocks to zs_refs to better reflect the new uses and added comment.